### PR TITLE
Docs cleanup

### PR DESCRIPTION
**Pull request by klecki**
There is still warning from notebooks which I don't know how to handle:
```
/dali/docs/examples/mxnet/mxnet-various-readers.ipynb:23: WARNING: File not found: 'examples'            
/dali/docs/examples/paddle/paddle-basic_example.ipynb:16: WARNING: File not found: 'examples/paddle/...' 
/dali/docs/examples/paddle/paddle-various-readers.ipynb:23: WARNING: File not found: 'examples'          
/dali/docs/examples/pytorch/pytorch-basic_example.ipynb:16: WARNING: File not found: 'examples/pytorch/..
/dali/docs/examples/pytorch/pytorch-various-readers.ipynb:23: WARNING: File not found: 'examples'        
/dali/docs/examples/serialization.ipynb:21: WARNING: File not found: 'examples'                          
/dali/docs/examples/tensorflow/tensorflow-plugin-tutorial.ipynb:32: WARNING: File not found: 'examples'  
/dali/docs/examples/tensorflow/tensorflow-various-readers.ipynb:23: WARNING: File not found: 'examples'  
```

#### Why we need this PR?
It cleans the formatting in docs and clears most of the warnings.

#### What happened in this PR?
 - Explain solution of the problem, new feature added.

The docs were formatted according to https://numpydoc.readthedocs.io/en/latest/format.html
and rst guide (which they use to be rendered). 

 - What was changed, added, removed?

Mostly adjust indents & insert new lines to allow sphinx to properly pick-up paragraphs, use `::` for blocks of code/formula, introduce next level of headings in compilation.rst, adjust conf.py to not use `_static`.

Inserted a lot of `.` at the end of the class docstrings, but didn't touch the kwargs docstrings which still miss them.

 - What is most important part that reviewers should focus on?

Mostly on formatting of `supported operations` page in docs.

 - Was this PR tested? How?

Local make html, should be validated with CI build for docs.

 - Were docs and examples updated, if necessary?

This updates the formatting of the docs, not the contents.

**JIRA TASK**: [DALI-1191]

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by JanuszL** [comment]
Regarding remaining warnings, we use `[examples](...)`. I don't know if this is the right thing to do. Maybe we can rework it?

File: dali/operators/color/brightness_contrast.cc [hunk 1]
```diff
@@ -25,9 +25,9 @@ using TheKernel = kernels::MultiplyAddCpu<Out, In, 3>;
 
 
 DALI_SCHEMA(BrightnessContrast)
-    .DocStr(R"code(Adhust the brightness and contrast of the image according to the formula:
+    .DocStr(R"code(Adhust the brightness and contrast of the image according to the formula::
```
Comment (banasraf) [inline dali/operators/color/brightness_contrast.cc]:
```suggestion
    .DocStr(R"code(Adjust the brightness and contrast of the image according to the formula::
```

**Comment by klecki** [comment]
!build

**Pull request merged**